File: scope104/src/main/java/com/petro/scope104/data/UserDataSource.java
package com.petro.scope104.data;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;

import com.petro.scope104.domain.entity.WorkerEntity;
import com.petro.scope104.presentation.list.Gender;

import java.util.List;

public interface UserDataSource {
    LiveData<List<WorkerEntity>> loadUsers(int pageNumber, int pageSize, @NonNull Gender gender, @NonNull List<String> countries);
    void saveUsers(List<WorkerEntity> users);
}

File: scope104/src/main/kotlin/com/petro/scope104/data/UserDataSource.kt
package com.petro.scope104.data

import com.petro.scope104.domain.entity.WorkerEntity
import com.petro.scope104.presentation.list.Gender
import io.reactivex.rxjava3.core.Observable

interface UserDataSource {
    fun loadUsers(
        pageNumber: Int,
        pageSize: Int,
        gender: Gender,
        countries: List<String>
    ): Observable<List<WorkerEntity>>

    fun saveUsers(users: List<WorkerEntity>)
}